metron-interface/metron-rest/src/test/java/org/apache/metron/rest/controller/AlertsProfileControllerIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user1ProfileJson + "]"));
+
+    this.mockMvc.perform(post(url).with(httpBasic(user2, password)).with(csrf())
+        .contentType(MediaType.parseMediaType("application/json;charset=UTF-8"))
+        .content(user2ProfileJson))
+        .andExpect(status().isCreated())
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user1ProfileJson));
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user1ProfileJson + "," + user2ProfileJson + "]"));
+
+    this.mockMvc.perform(delete(url + "/user1").with(httpBasic(admin, password)))
+        .andExpect(status().isOk());
+
+    this.mockMvc.perform(delete(url + "/user1").with(httpBasic(admin, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json(user2ProfileJson));
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
+        .andExpect(status().isOk())
+        .andExpect(
+            content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
+        .andExpect(content().json("[" + user2ProfileJson + "]"));
+
+    this.mockMvc.perform(delete(url + "/user2").with(httpBasic(admin, password)))
+        .andExpect(status().isOk());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user1, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url).with(httpBasic(user2, password)))
+        .andExpect(status().isNotFound());
+
+    this.mockMvc.perform(get(url + "/all").with(httpBasic(admin, password)))
```

Comment:
Yeah, minimally we should explain what's going on and what the pre-reqs are for a test case. I agree with us having existing monolithic tests, and this isn't the right place to redesign everything. I prefer not having a large waterfall of dependencies between the steps. Ttehere are side effects between the steps that just aren't immediately obvious. 

Side note - this is mocking the mvc architecture - is it spinning up anything expensive? Would there be much cost in having each test do the necessary setup in an `@Before` method or within the test method itself?

Even better than documentation, I'd prefer splitting off well-named methods that explicitly state what the step is doing, e.g. 
```
@Test
public void add_profile_makes_mike_happy() {
    assertFalse(checkProfileExists(user1, password));
    assertFalse(checkProfileExists(user2, password));
    addUserProfile(user1, password, profile1);
    addUserProfile(user2, password, profile2);
    assertTrue(checkProfileExists(user1, password));
    assertTrue(checkProfileExists(user2, password));
}
```

As it's currently laid out, I find it difficult to know what the tests are doing.

Thoughts?